src/sinks/sftp/config.rs

```diff
@@ -0,0 +1,186 @@
+use codecs::{encoding::Framer, JsonSerializerConfig, NewlineDelimitedEncoderConfig};
+use opendal::{layers::LoggingLayer, services::Sftp, Operator};
+use tower::ServiceBuilder;
+use vector_config::configurable_component;
+use vector_core::{
+    config::{AcknowledgementsConfig, DataType, Input},
+    sink::VectorSink,
+};
+
+use crate::{
+    codecs::{Encoder, EncodingConfigWithFraming, SinkType},
+    config::{GenerateConfig, SinkConfig, SinkContext},
+    sinks::{
+        opendal_common::*,
+        util::{
+            partitioner::KeyPartitioner, BatchConfig, BulkSizeBasedDefaultBatchSettings,
+            Compression,
+        },
+        Healthcheck,
+    },
+};
+
+/// Configuration for the `sftp` sink.
+#[configurable_component(sink("sftp", "Sftp."))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct SftpConfig {
+    /// The root path for Sftp.
+    ///
+    /// Must be a valid directory.
+    ///
+    /// The final file path is in the format of `{root}/{prefix}{suffix}`.
+    #[serde(default)]
+    pub root: String,
+
+    /// A prefix to apply to all keys.
+    ///
+    /// Prefixes are useful for partitioning objects, such as by creating a blob key that
+    /// stores blobs under a particular directory. If using a prefix for this purpose, it must end
+    /// in `/` to act as a directory path. A trailing `/` is **not** automatically added.
+    ///
+    /// The final file path is in the format of `{root}/{prefix}{suffix}`.
+    #[serde(default)]
+    #[configurable(metadata(docs::templateable))]
+    pub prefix: String,
+
+    /// The endpoint to connect to sftp.
+    #[serde(default)]
+    #[configurable(metadata(docs::examples = "127.0.0.1:22"))]
+    pub endpoint: String,
+
+    /// The user to connect to sftp.
+    #[serde(default)]
+    #[configurable(metadata(docs::examples = "ubuntu"))]
+    pub user: String,
+
+    /// The key path that sftp used to connect to sftp.
+    #[serde(default)]
+    #[configurable(metadata(docs::examples = "/path/to/ssh/key/path"))]
+    pub key: String,
```

Comment:
I would suggest naming this `ssh_key` so that the field name is more self-documenting.

```suggestion
    /// The path of the SSH key used to connect to the SFTP backend.
    #[serde(default)]
    #[configurable(metadata(docs::examples = "/path/to/ssh/key/path"))]
    pub ssh_key_file: String,
```